### PR TITLE
Publish draft of npm best practices guide

### LINKED ISSUE #2: In progress: npm best practices guide
I'm currently converting https://docs.google.com/document/d/1iSDXVv5KStA1LRY4n1tIQS3kp8BTu66HZf72Ev6PSpw/edit?usp=sharing&resourcekey=0-QhsKVXiSQZULr4gaPG1Ofw to markdown and will merge into the drafts directory.

### PR DESCRIPTION
**Pull request by jeffmendoza**
Step 2 of process.md. Publishing this draft to the drafts/ directory for
further refinement before proposal. Closes #2

### CODE REVIEW

File: drafts/npm.md
Comment (ljharb) [inline drafts/npm.md:56]: uppercase characters are supported for historical packages, but not for new ones, so it would be incorrect for tooling to ban them entirely.

File: drafts/npm.md
Comment (ljharb) [inline drafts/npm.md:109]: this seems disingenuous - it wouldn't have mattered if it was reproducible, because the developer intentionally sabotaged the project.

File: drafts/npm.md
Comment (ljharb) [inline drafts/npm.md:113]: i believe npm already does this by default, using an integrity hash

File: drafts/npm.md
Comment (ljharb) [inline drafts/npm.md:179]: this fact is why i believe it's a *harmful* practice for published packages to use dev-only lockfiles: because it gives them a falsely insulated experience, while their users are exposed to breakage and vulns in transitive deps.

File: drafts/npm.md
Comment (ljharb) [inline drafts/npm.md:195]: it's considered incredibly user-hostile to include a shrinkwrap file with a published package, because it blocks deduplication and *also* because it blocks the free flow of bugfixes and nonbreaking updates. This should be noted lest someone think this is a viable path for a package.

File: drafts/npm.md
Comment (ljharb) [inline drafts/npm.md:199]: i strongly disagree with this recommendation, for the reasons given above, and none of my 300+ packages (which represent well over 10% of npm's download traffic) include a lockfile, nor will they, because I think that would cause far more harm than it helps.

File: drafts/npm.md
Comment (ljharb) [inline drafts/npm.md:328]: this is strictly less secure than a human doing it locally, because it's a single factor, instead of the two-factor that human publishes require.

**Comment by jeffmendoza** [comment]
@ljharb fyi We'll be presenting the [process](https://github.com/ossf/package-manager-best-practices/blob/main/process.md) in tomorrow morning's WG meeting for how a doc will go from draft to rfc to published. At this point, I'm converting from a Google doc to markdown, and am not addressing comments. Once we have it in `drafts/` comments can be opened as issues, but the main period where authors will be addressing comments is during the 30-day review period. Thanks.

**Comment by ljharb** [comment]
ok, thanks for clarifying.

**Comment by jeffmendoza** [comment]
@laurentsimon Should all be here, ptal. I'll move over the existing comments as new issues once merged.

File: drafts/npm.md
Comment (laurentsimon) [reply drafts/npm.md:195]: we're not recommending it, just documenting that it exists. As @jeffmendoza said, this document will go go thru an embargo period to let folks comment. So we'll be able to tweak the language and point out important limitations.

File: drafts/npm.md
Comment (laurentsimon) [reply drafts/npm.md:199]: here we don't mean shrinkwrap files, we just mean a lock file used for _CI_ - which is not used during package installation and won't "blocks the free flow of bugfixes and nonbreaking updates". 

I agree the way it's written is confusing. Thanks for pointing this out.

File: drafts/npm.md
Comment (laurentsimon) [reply drafts/npm.md:328]: Good point. Let's split this into different sections for "manual" and "automated".

**Review by laurentsimon — APPROVED** [review]
Thanks!

File: drafts/npm.md
Comment (ljharb) [reply drafts/npm.md:199]: Yes, but I still think that such a dev-only lockfile is harmful, because it falsely insulates maintainers from problems their users will encounter.

**Pull request merged**
